docs/upgrading.md

```diff
@@ -32,12 +32,53 @@ For the Emissary executor to work properly, you must set up RBAC. See [workflow
 
 ### Archived Workflows on PostgreSQL
 
-This upgrade will transform an archived workflow column from type `json` to type `jsonb`.
-This will take some time if you have a lot of archived workflows.
+To improve performance, this upgrade will automatically transform the column used to store archived workflows from type `json` to type `jsonb` on controller start-up.
 This requires PostgreSQL version 9.4 or higher.
 
-You can perform this modification prior to upgrading with `alter table argo_archived_workflows alter column workflow set data type jsonb using workflow::jsonb`.
-This is considered safe to do whilst running version 3.5 as the column types are compatible.
+The migration involves obtaining an [ACCESS EXCLUSIVE](https://www.postgresql.org/docs/current/explicit-locking.html) lock on the `argo_archived_wokflows` table, which blocks all reads and writes until it has finished.
+For the vast majority of users, we anticipate this will take less than a minute, but it could take much longer if you have a large number of workflows (100,000+), or the average workflow size is high (100KB+).
+If you don't fall into one of those two categories, or if minimizing downtime isn't important to you, then you don't need to read any further.
```

Comment:
Can the following details be hidden in a collapsible section so that readers can expand to see details when needed?

Reply:
That's a good idea, but unfortunately markdownlint won't allow it:
```
 $ make docs-lint
GIT_COMMIT=bcf7184f39da7adaac85e38beed8bc7cdefa0448 GIT_BRANCH=docs-upgrade-jsonb GIT_TAG=untagged GIT_TREE_STATE=dirty RELEASE_TAG=false DEV_BRANCH=true VERSION=latest
KUBECTX=k3d-k3s-default DOCKER_DESKTOP=false K3D=true DOCKER_PUSH=false TARGET_PLATFORM=linux/amd64
RUN_MODE=local PROFILE=minimal AUTH_MODE=hybrid SECURE=false  STATIC_FILES=false ALWAYS_OFFLOAD_NODE_STATUS=false UPPERIO_DB_DEBUG=0 LOG_LEVEL=debug NAMESPACED=true
npm list -g markdownlint-cli@0.33.0 > /dev/null || npm i -g markdownlint-cli@0.33.0
# lint docs
markdownlint docs --fix --ignore docs/fields.md --ignore docs/executor_swagger.md --ignore docs/cli --ignore docs/walk-through/the-structure-of-workflow-specs.md
docs/upgrading.md:42:1 MD033/no-inline-html Inline HTML [Element: details]
docs/upgrading.md:43:1 MD033/no-inline-html Inline HTML [Element: summary]
```

and it seems no way to selectively disable that rule for `upgrading.md`: https://github.com/markdownlint/markdownlint/issues/16

Reply:
Apparently, it is possible to possible to allowlist `<details>`/`<summary>` in `.markdownlint.yaml`: https://github.com/markdownlint/markdownlint/pull/450

but we'd also need to convert the list to HTML, since Markdown lists aren't rendered inside a `<details>` element, which means all those elements would need to be added to the allowlist too. We might as well just disable that rule at that point

Reply:
There's an alternative syntax for collapsible blocks that `mkdocs` supports: https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks

but I tried that, and `markdownlint` doesn't like that either.

Reply:
Alright, this is fine then. Can we highlight this sentence?

Reply:
Sure, done in 4540b41c060dbc69d098ff7ece5a1d64d6a59b81